net/cryfs/Portfile

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+PortSystem                1.0
+PortGroup                 github 1.0
+PortGroup                 cmake 1.1
+PortGroup                 active_variants 1.1
+
+
+github.setup              cryfs cryfs 0.10.2
+github.tarball_from       releases
+
+license                   LGPL-3
+
+categories                net fuse
+platforms                 darwin
+
+maintainers               {@jolan78 openmaintainer}
+
+description               Cryptographic filesystem for the cloud
+long_description          CryFS encrypts your files, so you can safely store them anywhere.\
+                          It works well together with cloud services like Dropbox, iCloud,\
+                          OneDrive and others
+
+homepage                  https://www.cryfs.org/
+
+checksums                 rmd160  9abdb5e59263cf8ea978470f4c6a4086ddd8a860\
+                          sha256  004c7fdf537062e197aaa0bcdafd2682904a279802441bb6063cb72c1fba52d2\
+                          size    10187912
+
+extract.mkdir             yes
+
+depends_build             port:openssl\
+                          port:python27\
+                          port:boost\
+                          path:/usr/local/include/osxfuse/fuse.h:osxfuse
+
+depends_lib               port:curl\
+                          port:libomp\
+                          path:/usr/local/lib/libosxfuse.2.dylib:osxfuse
+
+cmake.build_type          Release
+universal_variant         no
+default_variants          +multithread_boost
+
+variant no_version_check description {Doesn't check online for updates and security vulnerabilities} {
+    configure.args-append -DCRYFS_UPDATE_CHECKS=off
+}
+
+variant static_boost description {Link boost statically (requires -no_static variant of boost)} {
+    require_active_variants boost {} no_static
+}
+
+if {![variant_isset static_boost]} {
+    depends_lib-append port:boost
+    configure.args-append -DBoost_USE_STATIC_LIBS=OFF
+}
+
+variant multithread_boost description {Use the default multithreaded variant of boost} {
+    configure.args-append    -DBoost_USE_MULTITHREADED=on
+}
+
+if {![variant_isset multithread_boost]} {
+    require_active_variants boost {} no_single
+}
+
+variant system_osxfuse description {Prefer osxfuse installed from osxfuse.github.io} {
+    configure.cxxflags-append  -I/usr/local/include
+    configure.args-append      -DFUSE_LIB_PATH=/usr/local/lib/
+}
+
+# install port's osxfuse if system_osxfuse variant is not selected
+if {![variant_isset system_osxfuse]} {
+    depends_lib-append port:osxfuse
+}
+
```

Comment:
This is quite a lot of variants, are all of them really useful? Could some of them just be made defaults?

Reply:
I tried to support the common build options for the project.

+no_version_check could be useful: version check could be seen as a security weakness.
+static_boost has the advantage of removing a runtime dependency at the cost of increased build time, as boost is precompiled with no_static. it may be removed.
-multithread_boost is not really useful and I will remove it.
+system_osxfuse is handy IMO as  many projects outside macports depends on osxfuse, and maport's version is slightly outdated.

I'd like to know what you think about these.

Reply:
resolving this change since +no_version_check is discussed below.